app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java

```diff
@@ -67,9 +63,9 @@ public final boolean blockExists(String owner, String key, boolean revoked) {
      * @inheritDoc
      */
     @Override
-    public final List<Block> addBlockToChain(Block block) {
+    public final List<Block> addBlockToChain(Block block) throws HashException {
         // Check if the block already exists
-        String owner = block.getOwner();
+        String owner = block.getOwner().getName();
```

Comment:
You should not expect a name. The Owner of a block is a user; otherwise it would be getOwnerName. It does not readability as getUser() could also mean the user that has the public key inside the block.